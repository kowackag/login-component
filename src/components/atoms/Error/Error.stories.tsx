import type { Meta, StoryObj } from "@storybook/react";

import { ErrorMessage } from "./Error";

const meta: Meta<typeof ErrorMessage> = {
	component: ErrorMessage,
};

export default meta;

type Story = StoryObj<typeof ErrorMessage>;

export const Default: Story = {
	args: {
		message: "some error",
	},
};
