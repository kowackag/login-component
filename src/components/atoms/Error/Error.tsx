export const ErrorMessage = ({ message }: { message: string }) => {
	return (
		<p data-testid="error" data-  className="absolute right-0 font-primaryMedium text-xs italic text-red-600">{message}</p>
	);
};
