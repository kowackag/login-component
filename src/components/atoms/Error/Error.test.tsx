import { expect, test } from 'vitest'
import { render, screen } from "@testing-library/react";
import { ErrorMessage } from './Error';

test("Error displays appropriate message", () => {
	render(<ErrorMessage message="some error"/>);

	const message = screen.getByText("some error");
	expect(message).toHaveTextContent("some error");
});