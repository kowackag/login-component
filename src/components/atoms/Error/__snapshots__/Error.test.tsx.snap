// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error > renders correctly 1`] = `
<div>
  <p
    class="absolute right-0 font-primaryMedium text-xs italic text-red-600"
    data-="true"
    data-testid="error"
  >
    some error
  </p>
</div>
`;
